File: slides/code/processor-class.ts
type TemplateArgs = {
   reader: Reader;
   writer: Writer;
   label: string;
};

export class TemplateProcessor extends
      Processor<TemplateArgs> {
   async init(this: TemplateArgs & this):
      Promise<void> {
      // Initialization code here e.g., setting up
      // connections or loading resources
   }

   async transform(this: TemplateArgs & this):
      Promise<void> {
      // Function to start reading channels.
   }

   async produce(this: TemplateArgs & this):
      Promise<void> {
      // Function to start the production of data,
      // starting the pipeline.
   }
}

File: slides/code/processor-class.py
@dataclass
class TemplateArgs(ProcessorArgs):
    reader: Reader
    writer: Writer
    label: str

class TemplateProcessor(Processor[TemplateArgs]):
    logger: Logger = getLogger('rdfc.TemplateProcessor')

    def __init__(self, args: TemplateArgs):
        super().__init__(args)

    async def init(self) -> None:
        # Initialization code here e.g., setting up
        # connections or loading resources
        pass

    async def transform(self) -> None:
        # Function to start reading channels.
        pass

    async def produce(self) -> None:
        # Function to start the production of data,
        # starting the pipeline.
        pass
